pandas/tests/window/test_rolling.py

```diff
@@ -130,29 +132,125 @@ def test_closed_fixed(closed, arithmetic_win_operators):
-    )
+    result = getattr(
+        df_fixed.rolling(2, closed=closed, min_periods=1),
+        func_name,
+    )()
+    expected = getattr(
+        df_time.rolling("2D", closed=closed, min_periods=1),
+        func_name,
+    )().reset_index(drop=True)
 
     tm.assert_frame_equal(result, expected)
 
 
-def test_closed_fixed_binary_col():
+@pytest.mark.parametrize(
+    "closed, window_selections",
+    [
+        (
+            "both",
+            [
+                [True, True, False, False, False],
+                [True, True, True, False, False],
+                [False, True, True, True, False],
+                [False, False, True, True, True],
+                [False, False, False, True, True],
+            ],
+        ),
+        (
+            "left",
+            [
+                [True, False, False, False, False],
+                [True, True, False, False, False],
+                [False, True, True, False, False],
+                [False, False, True, True, False],
+                [False, False, False, True, True],
+            ],
+        ),
+        (
+            "right",
+            [
+                [True, True, False, False, False],
+                [False, True, True, False, False],
+                [False, False, True, True, False],
+                [False, False, False, True, True],
+                [False, False, False, False, True],
+            ],
+        ),
+        (
+            "neither",
+            [
+                [True, False, False, False, False],
+                [False, True, False, False, False],
+                [False, False, True, False, False],
+                [False, False, False, True, False],
+                [False, False, False, False, True],
+            ],
+        ),
+    ],
+)
+def test_datetimelike_centered_selections(
+    closed, window_selections, arithmetic_win_operators
+):
+    # GH 34315
+    func_name = arithmetic_win_operators
+    df_time = DataFrame(
+        {"A": [0.0, 1.0, 2.0, 3.0, 4.0]}, index=date_range("2020", periods=5)
+    )
+
+    expected = DataFrame(
+        {"A": [getattr(df_time["A"].iloc[s], func_name)() for s in window_selections]},
+        index=date_range("2020", periods=5),
+    )
+
+    if func_name == "sem":
+        kwargs = {"ddof": 0}
+    else:
+        kwargs = {}
+
+    result = getattr(
+        df_time.rolling("2D", closed=closed, min_periods=1, center=True),
+        func_name,
+    )(**kwargs)
+
+    tm.assert_frame_equal(result, expected, check_dtype=False)
+
+
+def test_even_number_window_alignment():
+    # see discussion in GH 38780
+    s = Series(range(3), index=date_range(start="2020-01-01", freq="D", periods=3))
+
+    # behavior of index- and datetime-based windows differs here!
+    # s.rolling(window=2, min_periods=1, center=True).mean()
+
+    result = s.rolling(window="2D", min_periods=1, center=True).mean()
+
+    expected = Series([0.5, 1.5, 2], index=s.index)
+
+    tm.assert_series_equal(result, expected)
```

Comment:
@mroeschke I added this test to check for the behavior when operating with even sized windows. As mentioned by @sevberg this currently does behave differently than the classic index-based windows.

Is that what you had in mind [here](https://github.com/pandas-dev/pandas/pull/38780#issuecomment-772766767)?

Reply:
This test would also work with `N` instead of `D` as frequency, therefore I am not sure if this is what you meant when referring to nano seconds?

Reply:
Yes I believe this test exercises the difference in behavior which is good.

Reply:
Great, thanks!